src/WebJobs.Script.WebHost/WebHooks/DynamicWebHookReceiverConfig.cs

```diff
@@ -20,13 +22,8 @@ public Task<string> GetReceiverConfigAsync(string name, string id)
             // "id" will be the function name
             // we ignore the "name" parameter since we only allow a function
             // to be mapped to a single receiver
-            FunctionSecrets secrets = _secretManager.GetFunctionSecrets(id);
-            if (secrets != null)
-            {
-                return Task.FromResult(secrets.Key);
-            }
-
-            return null;
+            string functionSecret = _secretManager.GetFunctionSecrets(id).Values.FirstOrDefault();
```

Comment:
I may be missing something, but different WebHook producers/receivers use the key differently, and the receiver specific implementation handles validation against the secret value.
For example, some providers will use the key to sign the request, and the receiver uses the key returned here to validate that signature. Unless we want to force the use of a code with the URL and just return the matching secret (if found) to the WebHooks SDK, but that would still require us to pass some additional context here and passing the identifier instead would probably be cleaner, and make key management simpler in some cases.